File: leetcode/design/379_Design_Phone_Directory/codes/array.cpp
#include <vector>
using namespace std;

class PhoneDirectory {
    vector<bool> isSlotAvailable;

public:
    PhoneDirectory(int maxNumbers) {
        isSlotAvailable = vector<bool>(maxNumbers, true);
    }

    int get() {
        for (int i = 0; i < (int)isSlotAvailable.size(); ++i) {
            if (isSlotAvailable[i]) {
                isSlotAvailable[i] = false;
                return i;
            }
        }
        return -1;
    }

    bool check(int number) {
        return isSlotAvailable[number];
    }

    void release(int number) {
        isSlotAvailable[number] = true;
    }
};

/**
 * Your PhoneDirectory object will be instantiated and called as such:
 * PhoneDirectory* obj = new PhoneDirectory(maxNumbers);
 * int param_1 = obj->get();
 * bool param_2 = obj->check(number);
 * obj->release(number);
 */

File: leetcode/design/379_Design_Phone_Directory/codes/array.py
class PhoneDirectory:
    def __init__(self, maxNumbers: int):
        self.isSlotAvailable = [True] * maxNumbers

    def get(self) -> int:
        for i in range(len(self.isSlotAvailable)):
            if self.isSlotAvailable[i]:
                self.isSlotAvailable[i] = False
                return i
        return -1

    def check(self, number: int) -> bool:
        return self.isSlotAvailable[number]

    def release(self, number: int) -> None:
        self.isSlotAvailable[number] = True

# Your PhoneDirectory object will be instantiated and called as such:
# obj = PhoneDirectory(maxNumbers)
# param_1 = obj.get()
# param_2 = obj.check(number)
# obj.release(number)